common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/RemoteBlockPushResolver.java

```diff
@@ -648,14 +809,24 @@ public void onData(String streamId, ByteBuffer buf) throws IOException {
       // to disk as well. This way, we avoid having to buffer the entirety of every blocks in
       // memory, while still providing the necessary guarantee.
       synchronized (partitionInfo) {
-        Map<Integer, AppShufflePartitionInfo> shufflePartitions =
+        Map<Integer, Map<Integer, AppShufflePartitionInfo>> shuffleMergePartitions =
```

Comment:
BTW, it's better to rename this level map to `shuffleMergePartitionsByMergeId`. `shuffleMergePartitions` is better for the next level.